joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/utils/Scope.scala

```diff
@@ -104,6 +104,9 @@ class Scope(summary: Map[String, Seq[SymbolSummary]] = Map.empty)(implicit nextC
     methods
   }
 
+  def isSurroundedByMetaclassTypeDecl: Boolean =
+    stack.map(_.scopeNode.node).collectFirst { case td: NewTypeDecl if td.name.endsWith("<metaclass>") => td }.isDefined
```

Comment:
I'm not fully immersed in the implementation details of the scope classes. But I suspect this could go wrong with nested classes. (E.g. an anonymous class defined within a static method.) Should it be sth like this?

```suggestion
    stack.map(_.scopeNode.node).collectFirst { case td: NewTypeDecl => td }.exists(_ .name.endsWith(MetaTypeDeclExtension))
```